src/parser/toplevel.lisp

```diff
@@ -554,56 +544,214 @@ consume all attributes"))))
 
       (parse-expression form file))))
 
+;;; Packages
```

Comment:
Please rewrite this so that it matches the conventions used throughout the rest of the parser. The custom iterator implementation seems unnecessary.